### PR TITLE
fix copy effect not being discarded early enough

### LINKED ISSUE #12433: Bug: Shifting Woodland keeps the copied type and color in the graveyard
From the cube Discord:

[[Shifting Woodland]] copying [[Aftermath Analyst]] does not return itself when sacrificed with Analyst's ability.

image

It seems to keep the copied types (and color) in the graveyard.

image

### PR DESCRIPTION
**Pull request by Susucre**
close #12433

### CODE REVIEW

File: Mage/src/main/java/mage/abilities/effects/common/CopyEffect.java [hunk 1]
```diff
@@ -80,13 +80,10 @@ public boolean apply(Game game, Ability source) {
...
-            permanent = (Permanent) game.getLastKnownInformation(getSourceId(), Zone.BATTLEFIELD, source.getSourceObjectZoneChangeCounter());
-            // As long as the permanent is still in the LKI continue to copy to get triggered abilities to TriggeredAbilities for dies events.
-            if (permanent == null) {
-                discard();
-                return false;
-            }
+        if (permanent == null && !game.getShortLivingLKI(getSourceId(), Zone.BATTLEFIELD)) {
+            // As long as the permanent is still in the short living LKI continue to copy to get triggered abilities to TriggeredAbilities for dies events.
+            discard();
+            return false;
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/effects/common/CopyEffect.java:86]: If present in short living LKI, then don't you still need to copy from the LKI?

Comment (Susucre) [reply Mage/src/main/java/mage/abilities/effects/common/CopyEffect.java:86]: Oh right, I'll try to make a test that would have caught that for the day we eventually remove short lki.

Comment (Susucre) [reply Mage/src/main/java/mage/abilities/effects/common/CopyEffect.java]: Ok I couldn't have a test catching that, but now the permanent is set back to the LKI one if getShortLivingLKI returns true.

**Review by JayDi85 — APPROVED** [review]

**Review by xenohedron — APPROVED** [review]
(travis passes for PR build, ignore the verify failure in the branch build)

**Pull request merged**